server.js

```diff
@@ -27,6 +21,70 @@ app.get('/', (req, res) => {
   res.send('Hello world')
 })
 
+//all shows
+
+app.get('/shows', (req, res) => {
+  let shows = data.slice(0, 200)
+  const country = req.query.country
+  if(country) shows = data.filter(item => item.country === country)
+  res.json(shows)
+})
+
+//query for anything
+
+app.get('/shows/any', (req, res) => {
+  let shows = data.slice(0, 200)
+  const query = req.query.query
+  if(query) shows = data.filter(item => (
+    item.title === query ||
+    item.type === query ||
+    item.director === query ||
+    item.release_year === +query ||
+    item.country === query))
+  res.json(shows)
+})
+
+//all shows limited
+
+app.get('/shows/page', (req, res) => {
+  const page = req.query.page
+  let shows = data.slice(0, 50)
+  if(page) shows = data.slice(page*50, page*50 + 50)
+  res.json(shows)
+})
+
+//tv shows
+
+app.get('/shows/tvshows', (req, res) => {
+  const tvData = data.filter(item => item.type == 'TV Show')
+  res.json(tvData)
+}) 
+
+//movies
+
+app.get('/shows/movies', (req, res) => {
+  const movieData = data.filter(item => item.type == 'Movie')
+  res.json(movieData)
+}) 
+
+//shows from a specific year
+
+app.get('/shows/year/:year', (req, res) => {
+  const year = req.params.year
+  const showsFromYear = data.filter(item => item.release_year == year)
+  res.json(showsFromYear)
+})
+
+//show with a specific title
+
+app.get('/shows/title/:title', (req, res) => {
+  const titleID = req.params.title
+  const titleWithID = data.find(item => item.show_id == +titleID)
+  console.log(titleWithID)
+  if(!titleWithID) res.send('Not Found!!')
+  else res.json(titleWithID)
+})
```

Comment:
You've named the param title but expect an ID? 🤔 Change the naming here so it adds up :)

Reply:
Done